qiime/core/type/signature.py

```diff
@@ -94,10 +110,30 @@ def defaults(self):
             (name, spec.default) for name, spec in self.parameters.items()
             if spec.has_default()])
 
-    def _parse_signature(self, callable, inputs, parameters, outputs):
+    @property
+    def descriptions(self):
+        params = {**self.inputs, **self.parameters, **self.outputs}
```

Comment:
In the `NOVALUE` case, it would make sense for the CLI to store an empty string. But I'm fine with leaving as is. As far as ordering, that shouldn't matter to the CLI (as JSON doesn't have order anyways), and I don't see a tangible benefit for this helper (it doesn't need to be an authoritative source on order, and we probably don't want it to be used as such).